mne/time_frequency/psd.py

```diff
@@ -146,15 +153,37 @@ def compute_epochs_psd(epochs, picks=None, fmin=0, fmax=np.inf, n_fft=256,
         picks = pick_types(epochs.info, meg=True, eeg=True, ref_meg=False,
                            exclude='bads')
 
+    data = epochs.get_data()[:, picks]
+    if proj:
+        proj, _ = make_projector_info(raw.info)
+        if picks is not None:
+            data = np.dot(proj[picks][:, picks], data)
+        else:
+            data = np.dot(proj, data)
+
     logger.info("Effective window size : %0.3f (s)" % (n_fft / float(Fs)))
-    psds = []
+
     import matplotlib.pyplot as plt
-    parallel, my_psd, n_jobs = parallel_func(_compute_psd, n_jobs)
-    fig = plt.figure()  # threading will induce errors otherwise
-    out = parallel(my_psd(data[picks], fmin, fmax, Fs, n_fft, plt.psd,
-                          n_overlap, pad_to)
-                   for data in epochs)
-    plt.close(fig)
-    psds = [o[0] for o in out]
-    freqs = [o[1] for o in out]
-    return np.array(psds), freqs[0]
+    parallel, my_pwelch, n_jobs = parallel_func(_pwelch, n_jobs=n_jobs,
+                                                verbose=verbose)
+
+    psds = np.empty(data.shape[:-1] + (n_fft // 2 + 1,))
+    freqs = np.arange(psds.shape[-1]) * (Fs / n_fft)
+
+    for i_epoch, fepoch in enumerate(parallel(
+        my_pwelch(epoch, window_size=window_size,
+                  noverlap=n_overlap, nfft=n_fft, fs=Fs)
+                  for epoch in np.array_split(data, n_jobs))):
+
+        for i_channel, fchannel in enumerate(fepoch):
+            if i_epoch == 0:
+                psds[len(fepoch) * i_epoch + i_channel, :, :] = fchannel[1]
+            else:
+                psds[i_iter * i_epoch + i_channel, :, :] = fchannel[1]
```

Comment:
I've done it that way, because fepoch can have different sizes, and I needed the size of the previous iteration in order to have the right index in psds. 

e.g if I split my data of 100 epochs to 3 sets, the last set will have 34 epochs, so len(fepoch) won't be the same than previous to have the continuous index... Am I clear?

I'm looking again at it to see how can I avoid it.